plugins/module_utils/foreman_helper.py

```diff
@@ -471,6 +465,77 @@ def find_puppetclass(self, name, environment=None, params=None, failsafe=False,
     def find_puppetclasses(self, names, **kwargs):
         return [self.find_puppetclass(name, **kwargs) for name in names]
 
+    def scope_for(self, key):
+        return {'{0}_id'.format(key): self.lookup_entity(key)['id']}
+
+    def set_entity(self, key, entity):
+        self.foreman_params[key] = entity
+        self.foreman_spec[key]['resolved'] = True
+
+    def lookup_entity(self, key):
```

Comment:
```suggestion
    def lookup_entity(self, key='entity'):
```

I think we can default to looking up the "main" entity?

Reply:
even if at this level of module inheritance 'entity' isn't even a thing?

Reply:
:woman_shrugging: 

We can also override it in `ForemanEntityAnsibleModule`, it just felt weird to read `lookup_entity('entity')` so often in the diff. But you can also just call me crazy and I'll shut up :)